Add support for D4A/Draupnir For All to the playbook.

FSG-Cat opened the pull request:
This PR adds Draupnir for all support to the playbook. 

This is my first role that uses an AS and my first role that has a database involved so i took inspiration from multiple roles, Slack and AS Discord are roles that have inspired this one in parts. But this is also a straight clone of Draupnir in as many places as possible as well Draupnir For All and Draupnir are the same thing in a lot of ways. Since its the Appservice operating mode of Draupnir.

If this PR is merged this will resolve #3143 

The State of the PR as of when it was marked Ready for review is that the PR works. It deploys and you can succesfully configure a D4A. 

Edit: D4A is pinned to develop due to that there are patches in main currently waiting for the next release that are REQUIRED to make D4A work in mdad. A few hours of work together with MTRNord maintainer of the only D4A deployment besides this capability in the playbook and Gnuxie maintainer of Draupnir it self where involved in this and we could not get it working without the patches. (It works on Kubernetes without the patches for some reason but not on Docker.)

FSG-Cat committed:
Draupnir for all Role

FSG-Cat committed:
Draupnir for all Documentation

FSG-Cat committed:
Pin D4A to Develop until D4A patches are in a release.

spantaleev commented on docs/configuring-playbook-appservice-draupnir-for-all.md:
```diff
@@ -0,0 +1,87 @@
+
+The playbook can install and configure the [Draupnir](https://github.com/the-draupnir-project/Draupnir) moderation tool for you in appservice mode.
+
+Appservice mode can be used together with normal Draupnir or independently. 
```
```suggestion
Appservice mode can be used together with the regular [Draupnir bot](configuring-playbook-bot-draupnir.md) or independently. 
```

spantaleev replied:
Also, adding some information on pros/cons to using bot vs appservice, would be nice.

The appservice is more multitenant? It's also not rate-limitted? Etc.

FSG-Cat replied:
Appservice is Multitennant yes and Ratelimit Exempt just like a Normal Draupnir should be. Will add this to the documentation.

FSG-Cat committed:
Update D4A Docs to mention pros and cons of D4A mode compared to normal

FSG-Cat committed:
Change Documentation to mention a fixed simpler provisioning flow.

Use of /plain allows us to bypass the bugs encountered during the development of this role with clients attempting to escape our wildcards causing the grief that led to using curl.

This reworded commit does still explain you can automatically inject stuff into the room if you wanted to.

Gnuxie commented on docs/configuring-playbook-appservice-draupnir-for-all.md:
```diff
@@ -0,0 +1,93 @@
+
+Appservice mode can be used together with the regular [Draupnir bot](configuring-playbook-bot-draupnir.md) or independently.
+
+D4A compared to Draupnir Normal. 
```
Should you be calling this "normal"? Is that what it's called internal to mdad? it's probably better to call it "bot mode" so it's consistent with the way Draupnir's documentation refers to the same thing.

Gnuxie commented on docs/configuring-playbook-appservice-draupnir-for-all.md:
```diff
@@ -0,0 +1,93 @@
+
+Draupnir for all does also not support external tooling like [MRU](https://mru.rory.gay) as it cant access Draupnirs user account like is needed to use it. 
+
+Last downside of Draupnir for all is the state of the Admin side of the user experience being Alpha quality. This flaw is not in regular Draupnir and is mostly not in the provisioned Draupnirs either.
```
This should be the first thing they read in relation to the appservice mode, not the last thing.

Gnuxie replied:
"The administrative functions for managing the appservice are alpha quality and very limited. However, the experience of using an appservice provisioned Draupnir is on par with the experience of using Draupnir from bot mode."

FSG-Cat committed:
Emphasise the State of D4A mode

spantaleev committed:
Link to Draupnir-for-all docs and tweak the docs some

spantaleev committed:
Link to Draupnir-for-all from Draupnir documentation page

spantaleev committed:
Announce Draupnir-for-all
